pkg/ottl/expression.go

```diff
@@ -245,6 +246,45 @@ func (g StandardFloatGetter[K]) Get(ctx context.Context, tCtx K) (float64, error
 	}
 }
 
+// FunctionGetter is a Getter that must return an ExprFunc.
+type FunctionGetter[K any] interface {
+	Get(ctx context.Context, iArgs Arguments) (Expr[K], error)
+}
+
+// StandardFunctionGetter is a basic implementation of FunctionGetter
+type StandardFunctionGetter[K any] struct {
+	fCtx FunctionContext
+	fact Factory[K]
+}
+
+// Get retrieves an ExprFunc value.
+// If the value is not an ExprFunc a new Error is returned.
+// If there is an error getting the value it will be returned.
+func (g StandardFunctionGetter[K]) Get(_ context.Context, iArgs Arguments) (Expr[K], error) {
+	if g.fact == nil {
+		return Expr[K]{}, fmt.Errorf("undefined function")
+	}
+	fArgs := g.fact.CreateDefaultArguments()
+	fArgsVal := reflect.ValueOf(fArgs).Elem()
+	iArgsVal := reflect.ValueOf(iArgs).Elem()
+	if fArgsVal.NumField() != iArgsVal.NumField() {
+		return Expr[K]{}, fmt.Errorf("incorrect number of arguments. Expected: %d Received: %d", fArgsVal.NumField(), iArgsVal.NumField())
+	}
+	for i := 0; i < fArgsVal.NumField(); i++ {
+		field := iArgsVal.Field(i)
+		_, err := checkForArgErrors(i, fArgsVal)
```

Comment:
We need to perform this check for `iArgsVal` as well. Right now we're relying on the fact that struct field ordering is stable in Go to get the right argument index. In reality this code will work in most cases, but we decided we wanted to make argument ordering explicit and require the `ottlarg` tag.

Reply:
validating iArgsVal seems to return this error
```no `ottlarg` struct tag on Arguments field "Input"```
eventhough this is defined like so
```
type ReplacementFuncArgs[K any] struct {
	Input ottl.StringGetter[K] `ottlarg:"0"`
}
```

Reply:
Unfortunately nothing is jumping out at me for why this wouldn't work. I can try to take a look on Monday if necessary.

Reply:
oh nvm, it was an issue with my test input